src/main/kotlin/com/autonomousapps/tasks/classAnalysisTasks.kt

```diff
@@ -177,54 +184,66 @@ abstract class ClassListAnalysisTask @Inject constructor(
 
   @TaskAction fun action() {
     // Output
-    val reportFile = output.getAndDelete()
-    val reportPrettyFile = outputPretty.getAndDelete()
+    val reportFile = output
+    val reportPrettyFile = outputPretty
+
+    val inputClassFiles = project.objects.setProperty(FileSystemLocation::class.java)
```

Comment:
what's up with this? I think we shouldn't do this because, at the very least, calling `projects` definitely breaks the configuration cache contract. If we need a SetProperty as an input, we can do it at the class level.